streams/src/main/java/org/apache/kafka/streams/state/internals/MemoryLRUCache.java

```diff
@@ -88,17 +81,9 @@ public String name() {
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public void init(ProcessorContext context, StateStore root) {
-        if (loggingEnabled) {
-            context.register(root, true, new StateRestoreCallback() {
-
-                @Override
-                public void restore(byte[] key, byte[] value) {
-                    put(serdes.keyFrom(key), serdes.valueFrom(value));
-                }
-            });
-
-        }
+        // do nothing
```

Comment:
Most of its initialization is done in constructor, this `init` function is only related to context during instantiation time.